lib/Cake/Network/CakeResponse.php

```diff
@@ -1464,7 +1465,7 @@ protected function _sendFile($file, $range) {
 		$file->open('rb');
 
 		$end = $start = false;
-		if ($range) {
+		if (is_array($range) && !empty($range)) {
```

Comment:
You are changing behavior again:
If the range is not an array, the whole functionality here differs now instead of blowing up as it should.

As discusses, please revert fully.

Reply:
Again: 2.x is in maintenance mode, security and critical bugfixes only.
No functional changes for the sake of changing anymore, please, to avoid further accidental regressions.

Reply:
Use `if ($range && is_array($range)) {` then to fit the guidelines.

Reply:
You are still wrong about what it means: in PHP it doesnt mean true, but not empty in the context of arrays. Your code is cloaking which is 10x worse. Let's please stop this discussion.
The more important question is: Why the array check now? What happens if it is not an array?
Then it would silently continue doing sth different then it did before.

Reply:
> in PHP it doesnt mean true, but not empty in the context of arrays.

I think you are wrong about that. From PHP documentation:

> As described in the section about expressions, expression **is evaluated to its Boolean value**.
http://php.net/manual/en/control-structures.if.php

> When necessary, an expression is **automatically converted to boolean**.
http://php.net/manual/en/language.expressions.php

> To explicitly convert a value to boolean, use the (bool) or (boolean) casts. However, in most cases the cast is unnecessary, since a value will be **automatically converted** if an operator, function or control structure requires a boolean argument.
http://php.net/manual/en/language.types.boolean.php#language.types.boolean.casting

So event when `$var` is an array `if ($var)` is evaluated as `if var is true` according to PHP documentation. `if ($var)` is equivalent to `if ($var == true)` - non-strict comparison to a boolean.

---
> The more important question is: Why the array check now? What happens if it is not an array?

If `$range` is not an array but `1` or `true` then `$start` and `$end` are assigned `null` values although `false` values are expected in such scenario - take a look [here](http://sandbox.onlinephpfunctions.com/code/6623b50ebc865963924eb4fa8730c38bd2d4d922) again and compare it with the [correct](http://sandbox.onlinephpfunctions.com/code/f1307e1bdb883a91cb6a741342cc20a535ba8e66) implementation please.

Reply:
Then `if ($range && is_array($range)) {` is the correct code here.

Reply:
Okay, but when you claimed that `in PHP it doesnt mean true` you de-railed the discussion to PHP level yourself.

Reply:
TLDR:  Don't make any changes that can potentially change the way code behaves. 2.x is **done**.